source/common/network/utility.cc

```diff
@@ -299,6 +308,14 @@ Address::InstanceConstSharedPtr Utility::getIpv6AnyAddress() {
   return any;
 }
 
+Address::InstanceConstSharedPtr Utility::getNullLocalAddress(const Address::Instance& address) {
+  if (address.type() == Address::Type::Ip && address.ip()->version() == Address::IpVersion::v6) {
+    return Utility::getIpv6AnyAddress();
+  }
+  // Default to IPv4 any address.
```

Comment:
Currently, the code expects that local address is not nullptr even for address type Pipe.